### DISCUSSION TITLE
Static/skipping via librespot and shairport-sync

### DISCUSSION THREAD

**Discussion by aeleeds** [snapcast/snapcast#1033] [Q&A]
Hi everyone,
I've got a snapcast setup running on a couple of raspberry pis (server is a 4, client is a 3b, both running buster) outputting via USB to an integrated amp/dac. Audio is clean and skipless coming from mpd, but unlistenably and continuously staticy/skippy via librespot and shairport-sync. 

My sources are configured:

```
source = pipe:///tmp/snapfifo?name=default
source = librespot:///librespot?name=Spotify&username=<myusr>&password=<mypass>&devicename=Snapcast&autoplay=false
source = airplay:///shairport-sync?name=Airplay&devicename=Snapcast&port=5000
```

And everything else in snapcast.conf is at default. Snapserver's default file is empty, and Snapclient's only default setting is for the soundcard and /etc/snapclient.conf is empty.

The wifi connection is strong. 

Sound is handled via ALSA.

What am I doing wrong? What info can I send you to help?

**Comment by artmg** [comment]
I have a config similar to yours, although I have `&bitrate=320` on librespot. 

Skipping during playback is likely to be something making those digital sound bits slow down - whether that's network or cpu could depend on your setup. I have the fortune of having run ethernet between my snapcast Pis and my network switch, but Wifi signal is not the only measure of network issues.

Don't forget that MPD generates the audio locally on the snapserver, whereas airplay has the audio streaming across the network, and librespot has the internet to rely on. Have you tried airplaying some locally stored sound files? If your problem is Spotify->Airplay->Snapserver AND Spotify->LibreSpot->Snapserver then I'd suspect your internet connection to Spotify. 

The first place to start with diagnostics is the snapcast classic: stop the services and run the server and client from the console in debug logging mode, e.g.: `snapclient --logfilter debug` or `snapserver --logging.filter debug`

You might we see something there that indicates where your issues are. Otherwise restart the services and monitor the CPU usage with `top` whilst playing, to see if it's CPU that is your bottleneck. You can do some basic network latency (delay) and bandwidth (throughput) tests using `ping` and `iperf` - try during both silence and playback.

Good luck!
